cupy/cuda/thrust.pyx

```diff
@@ -12,45 +12,47 @@ from cupy.cuda cimport common
 ###############################################################################
 
 cdef extern from "../cuda/cupy_thrust.h" namespace "cupy::thrust":
-    void _sort[T](void *start, ptrdiff_t num)
+    void _sort[T](void *start, size_t ndim, size_t *shape)
     void _argsort[T](size_t *idx_start, void *data_start, size_t num)
 
 
 ###############################################################################
 # Python interface
 ###############################################################################
 
-cpdef sort(dtype, size_t start, size_t num):
-    cdef void* ptr
-    cdef Py_ssize_t n
+cpdef sort(dtype, size_t start, size_t ndim, size_t shape):
```

Comment:
How about uinsg `vector.vector[Py_ssize_t]& shape`?
Please see [this code](https://github.com/cupy/cupy/blob/master/cupy/core/internal.pyx#L17)